[WebGPU] Move prepareForDisplay() from PresentationContext to CompositorIntegration

litherum opened the pull request:
#### b54dc9debcec25e8e59b943c5e1afc863e181477
<pre>
[WebGPU] Move prepareForDisplay() from PresentationContext to CompositorIntegration
<a href="https://bugs.webkit.org/show_bug.cgi?id=251564">https://bugs.webkit.org/show_bug.cgi?id=251564</a>
rdar://104942364

Reviewed by Dean Jackson.

This patch starts the data flow for WebGPU compositing that I&apos;ve been working toward. Recall that we
have 2 related objects: 1) CompositorIntegration, which is our own object that has no WebGPU equivalent,
that we can use to add any methods we want, and 2) PresentationContext, which represents a
GPUCanvasContext (and a WGPUSurface/WGPUSwapChain pair).

In the new flow, whenever the canvas context is reconfigured, we synchronously tell the GPU process and
receive new IOSurfaces. This way, we don&apos;t have to send an IOSurface each frame, because the web process
already has all the IOSurfaces in the swap chain.

When creating a CompositorIntegration/PresentationContext pair, we do a handshake between them so
CompositorIntegration can call into PresentationContext and into the WGPUPresentationContext. This
handshake involves some blocks that get passed around, so these objects can call each others&apos; methods
without adding new API to the classes. The ownership model is that the CompositorIntegrationImpl
contains a strong ref to the PresentationContextImpl, and a strong ref to the WGPUPresentationContext
via the capture scope of a block it owns. I think this is the right ownership model: the
CompositorIntegration is conceptually a higher-level object than the PresentationContext, so a ref from
CompositorIntegration to PresentationContext seems like the right direction.

GPUCanvasContextCocoa::configure():
1. Calls CompositorIntegration::recreateRenderBuffers(). This synchronously recreates the IOSurfaces.
Not only does CompositorIntegration::recreateRenderBuffers() synchronously recreate the IOSurfaces, but
it also notifies the PresentationContext that new IOSurfaces were created. The PresentationContext
remembers the IOSurfaces, because...
2. GPUCanvasContextCocoa::configure() then calls PresentationContext::configure(). This is where all the
details of the texture objects get passed in. At this point, PresentationContext creates internal
textures wrapping each of the IOSurfaces, and retains them.

Because we eagerly create the textures, most of the rest of PresentationContext is very simple; all it
has to do is track which texture is the current one.

Similarly, because the web process receives all of the backing IOSurfaces ahead of time when the canvas
got reconfigured, GPUCanvasContext doesn&apos;t have to do much beyond tracking which IOSurface is the
current one.

GPUCanvasContextCocoa::prepareForDisplay() calls CompositorIntegration::prepareForDisplay() which just
calls PresentationContext::present(). This is not quite enough, because prepareForDisplay() needs to
synchronously wait until all work has been scheduled, so we don&apos;t insert an incomplete frame into the
CoreAnimation layer tree. This improvement is being tracked at
<a href="https://bugs.webkit.org/show_bug.cgi?id=250993.">https://bugs.webkit.org/show_bug.cgi?id=250993.</a>

This patch deletes PresentationConfiguration, because it now is exactly identical to CanvasConfiguration,
and there&apos;s no need to have 2 configuration structs with the same contents and same purpose.

There&apos;s only really one more piece to this WebGPU compositing work, and that is to use
PresentationContext::getCurrentTexture() instead of Device::createSurfaceTexture(). I&apos;ll do that in an
upcoming patch.

* Source/WebCore/Modules/WebGPU/GPUCompositorIntegration.cpp:
(WebCore::GPUCompositorIntegration::recreateRenderBuffers const):
(WebCore::GPUCompositorIntegration::prepareForDisplay):
(WebCore::GPUCompositorIntegration::getRenderBuffers const): Deleted.
* Source/WebCore/Modules/WebGPU/GPUCompositorIntegration.h:
* Source/WebCore/Modules/WebGPU/GPUPresentationConfiguration.h: Removed.
* Source/WebCore/Modules/WebGPU/GPUPresentationContext.cpp:
(WebCore::GPUPresentationContext::configure):
* Source/WebCore/Modules/WebGPU/GPUPresentationContext.h:
* Source/WebCore/PAL/PAL.xcodeproj/project.pbxproj:
* Source/WebCore/PAL/pal/CMakeLists.txt:
* Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUCompositorIntegrationImpl.cpp:
(PAL::WebGPU::CompositorIntegrationImpl::prepareForDisplay):
(PAL::WebGPU::CompositorIntegrationImpl::recreateRenderBuffers):
(PAL::WebGPU::CompositorIntegrationImpl::getRenderBuffers): Deleted.
(PAL::WebGPU::CompositorIntegrationImpl::recreateIOSurfaces): Deleted.
* Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUCompositorIntegrationImpl.h:
* Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUConvertToBackingContext.h:
* Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUImpl.cpp:
(PAL::WebGPU::GPUImpl::createPresentationContext):
* Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUPresentationContextImpl.cpp:
(PAL::WebGPU::PresentationContextImpl::configure):
(PAL::WebGPU::PresentationContextImpl::unconfigure):
* Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUPresentationContextImpl.h:
* Source/WebCore/PAL/pal/graphics/WebGPU/WebGPUCompositorIntegration.h:
* Source/WebCore/PAL/pal/graphics/WebGPU/WebGPUPresentationConfiguration.h: Removed.
* Source/WebCore/PAL/pal/graphics/WebGPU/WebGPUPresentationContext.h:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/html/canvas/GPUCanvasContextCocoa.cpp:
(WebCore::getCanvasWidth):
(WebCore::getCanvasHeight):
(WebCore::GPUCanvasContextCocoa::GPUCanvasContextCocoa):
(WebCore::GPUCanvasContextCocoa::reshape):
(WebCore::GPUCanvasContextCocoa::configure):
(WebCore::GPUCanvasContextCocoa::unconfigure):
(WebCore::GPUCanvasContextCocoa::getCurrentTexture):
(WebCore::GPUCanvasContextCocoa::prepareForDisplay):
(WebCore::getCanvasSizeAsIntSize): Deleted.
(WebCore::GPUCanvasContextCocoa::unconfigurePresentationContextIfNeeded): Deleted.
(WebCore::GPUCanvasContextCocoa::configurePresentationContextIfNeeded): Deleted.
* Source/WebCore/html/canvas/GPUCanvasContextCocoa.h:
* Source/WebGPU/WebGPU/PresentationContext.h:
* Source/WebGPU/WebGPU/PresentationContext.mm:
(WebGPU::PresentationContext::unconfigure):
(wgpuSwapChainRelease):
* Source/WebGPU/WebGPU/PresentationContextCoreAnimation.h:
* Source/WebGPU/WebGPU/PresentationContextCoreAnimation.mm:
(WebGPU::PresentationContextCoreAnimation::unconfigure):
* Source/WebGPU/WebGPU/PresentationContextIOSurface.h:
(WebGPU::PresentationContextIOSurface::create): Deleted.
(WebGPU::PresentationContextIOSurface::displayBuffer const): Deleted.
(WebGPU::PresentationContextIOSurface::drawingBuffer const): Deleted.
* Source/WebGPU/WebGPU/PresentationContextIOSurface.mm:
(WebGPU::PresentationContextIOSurface::create):
(WebGPU::PresentationContextIOSurface::PresentationContextIOSurface):
(WebGPU::PresentationContextIOSurface::displayBuffer const):
(WebGPU::PresentationContextIOSurface::drawingBuffer const):
(WebGPU::PresentationContextIOSurface::renderBuffersWereRecreated):
(WebGPU::PresentationContextIOSurface::configure):
(WebGPU::PresentationContextIOSurface::unconfigure):
(WebGPU::PresentationContextIOSurface::present):
(WebGPU::PresentationContextIOSurface::getCurrentTexture):
(WebGPU::PresentationContextIOSurface::getCurrentTextureView):
(wgpuSurfaceCocoaCustomSurfaceGetDisplayBuffer):
(wgpuSurfaceCocoaCustomSurfaceGetDrawingBuffer):
(WebGPU::optionsFor32BitSurface): Deleted.
(WebGPU::createIOSurface): Deleted.
(WebGPU::createSurfaceFromDescriptor): Deleted.
(WebGPU::PresentationContextIOSurface::nextDrawable): Deleted.
* Source/WebGPU/WebGPU/Texture.h:
* Source/WebGPU/WebGPU/Texture.mm:
(WebGPU::Texture::usage):
(WebGPU::Device::createTexture):
(WebGPU::usage): Deleted.
* Source/WebGPU/WebGPU/WebGPUExt.h:
* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteCompositorIntegration.cpp:
(WebKit::RemoteCompositorIntegration::recreateRenderBuffers):
(WebKit::RemoteCompositorIntegration::prepareForDisplay):
(WebKit::RemoteCompositorIntegration::getRenderBuffers): Deleted.
* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteCompositorIntegration.h:
* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteCompositorIntegration.messages.in:
* Source/WebKit/GPUProcess/graphics/WebGPU/RemotePresentationContext.cpp:
(WebKit::RemotePresentationContext::configure):
* Source/WebKit/GPUProcess/graphics/WebGPU/RemotePresentationContext.h:
* Source/WebKit/GPUProcess/graphics/WebGPU/RemotePresentationContext.messages.in:
* Source/WebKit/Scripts/webkit/messages.py:
(headers_for_type):
* Source/WebKit/Shared/WebGPU/WebGPUConvertFromBackingContext.h:
* Source/WebKit/Shared/WebGPU/WebGPUConvertToBackingContext.h:
* Source/WebKit/Shared/WebGPU/WebGPUPresentationConfiguration.cpp: Removed.
* Source/WebKit/Shared/WebGPU/WebGPUPresentationConfiguration.h: Removed.
* Source/WebKit/Shared/WebGPU/WebGPUPresentationConfiguration.serialization.in: Removed.
* Source/WebKit/Sources.txt:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteCompositorIntegrationProxy.cpp:
(WebKit::WebGPU::RemoteCompositorIntegrationProxy::recreateRenderBuffers):
(WebKit::WebGPU::RemoteCompositorIntegrationProxy::prepareForDisplay):
(WebKit::WebGPU::RemoteCompositorIntegrationProxy::getRenderBuffers): Deleted.
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteCompositorIntegrationProxy.h:
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemotePresentationContextProxy.cpp:
(WebKit::WebGPU::RemotePresentationContextProxy::configure):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemotePresentationContextProxy.h:

Canonical link: <a href="https://commits.webkit.org/259778@main">https://commits.webkit.org/259778@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a956a5c411c44c9a55aa06cb9dc95256d5d3c7d9

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/105821 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/14889 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/38669 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/115019 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/175147 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/109723 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/16308 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/6088 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/98053 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/114800 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/111572 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/12405 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/95377 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/39913 "Failed to checkout and rebase branch from PR 9498") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/109267 "Passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/94269 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/27018 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/81596 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/8160 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/28370 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/8642 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/4959 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/14269 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/47918 "Passed tests") | | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6766 "Failed to push commit to Webkit repository") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/10199 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

mwyrzykowski commented on Source/WebCore/html/canvas/GPUCanvasContextCocoa.cpp:
```diff
@@ -188,10 +205,16 @@ RefPtr<GraphicsLayerContentsDisplayDelegate> GPUCanvasContextCocoa::layerContent
+
+    m_layerContentsDisplayDelegate->setDisplayBuffer(m_configuration->renderBuffers[m_configuration->frameCount]);
+    m_compositingResultsNeedsUpdating = false;
+    m_configuration->frameCount = (m_configuration->frameCount + 1) % m_configuration->renderBuffers.size();
```
~~Is this always correct? What if in the case of triple buffering when rendering frame N, the render for frame N-2 is complete prior to the frame for render N-1? Shouldn't we choose the first available frame instead of round-robining?~~

litherum replied:
Isn't it bad to show frames out of order to the user?

litherum replied:
Do you think you could describe the frame ordering strategy you have in mind in more detail?

mwyrzykowski replied:
We should always present the frames to the user in order but this is independent of the order we render to the triple buffers, I can draw an image

mwyrzykowski replied:
I'm going to contradict what I wrote earlier. I think the issue is in the case of triple buffering, won't this mean when we display frame N, then we will render to frame N+2. It adds a latency of one frame

litherum replied:
Upon further discussion, I've filed https://bugs.webkit.org/show_bug.cgi?id=251625 and we're going to investigate this asynchronously.

mwyrzykowski commented on Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUCompositorIntegrationImpl.cpp:
```diff
@@ -42,38 +42,38 @@ CompositorIntegrationImpl::CompositorIntegrationImpl(ConvertToBackingContext& co
-    return m_renderBuffers.map([] (const auto& renderBuffer) {
-        return MachSendRight::adopt(IOSurfaceCreateMachPort(renderBuffer.get()));
-    });
+    // FIXME: https://bugs.webkit.org/show_bug.cgi?id=250993 Wait for the results to be fully drawn
```
can we have this member function take a completion handler as it will eventually need to be asynchronous?

mwyrzykowski commented on Source/WebGPU/WebGPU/PresentationContextIOSurface.mm:
```diff
@@ -27,87 +27,123 @@
 {
-    nextDrawable();
+    ASSERT(m_ioSurfaces.count == m_renderBuffers.size());
+    m_currentIndex = (m_currentIndex + 1) % m_renderBuffers.size();
```
this is fine for double buffering but like the other comment, in general triple buffering I think usually the free back buffer is chosen which is not necessarily the next one in the round robin

litherum replied:
(As above, investigating more asynchronously)

mwyrzykowski commented on Source/WebKit/GPUProcess/graphics/WebGPU/RemoteCompositorIntegration.messages.in:
```diff
@@ -25,8 +25,9 @@
-void GetRenderBuffers() -> (Vector<MachSendRight> renderBuffers) Synchronous NotStreamEncodableReply
+void RecreateRenderBuffers(int width, int height) -> (Vector<MachSendRight> renderBuffers) Synchronous NotStreamEncodableReply
 #endif
+void PrepareForDisplay() -> (bool dummy) Synchronous // FIXME: https://bugs.webkit.org/show_bug.cgi?id=250993 Wait for the results to be fully drawn
```
should you add `NotStreamEncodableReply` since that will be needed when we wait for the results to be drawn?

litherum replied:
Will it be needed? None of the data structures passed in the message (either sent or replied) are unable to be represented in raw memory. I thought this was why the only message that has NonStreamEncodable is the message which replies with a MachPort (because a Mach port cannot be represented in raw memory)

mwyrzykowski replied:
`NotStreamEncodableReply` is a workaround for asynchronous replies for synchronous messages, I will discuss offline

mwyrzykowski reviewed (approved):
This is fine but what is the benefit to having all the MachSendRight instances available in the web process as opposed to having one return by prepareForDisplay?

`sizeof(MachSendRight) == sizeof(unsigned int) + struct alignment` so sending the MachSendRight from the GPU process every frame was not really a heavy operation.

By sending them all, over we need more book-keeping

litherum committed:
[WebGPU] Move prepareForDisplay() from PresentationContext to CompositorIntegration
https://bugs.webkit.org/show_bug.cgi?id=251564
rdar://104942364

Reviewed by Dean Jackson.

This patch starts the data flow for WebGPU compositing that I've been working toward. Recall that we
have 2 related objects: 1) CompositorIntegration, which is our own object that has no WebGPU equivalent,
that we can use to add any methods we want, and 2) PresentationContext, which represents a
GPUCanvasContext (and a WGPUSurface/WGPUSwapChain pair).

In the new flow, whenever the canvas context is reconfigured, we synchronously tell the GPU process and
receive new IOSurfaces. This way, we don't have to send an IOSurface each frame, because the web process
already has all the IOSurfaces in the swap chain.

When creating a CompositorIntegration/PresentationContext pair, we do a handshake between them so
CompositorIntegration can call into PresentationContext and into the WGPUPresentationContext. This
handshake involves some blocks that get passed around, so these objects can call each others' methods
without adding new API to the classes. The ownership model is that the CompositorIntegrationImpl
contains a strong ref to the PresentationContextImpl, and a strong ref to the WGPUPresentationContext
via the capture scope of a block it owns. I think this is the right ownership model: the
CompositorIntegration is conceptually a higher-level object than the PresentationContext, so a ref from
CompositorIntegration to PresentationContext seems like the right direction.

GPUCanvasContextCocoa::configure():
1. Calls CompositorIntegration::recreateRenderBuffers(). This synchronously recreates the IOSurfaces.
Not only does CompositorIntegration::recreateRenderBuffers() synchronously recreate the IOSurfaces, but
it also notifies the PresentationContext that new IOSurfaces were created. The PresentationContext
remembers the IOSurfaces, because...
2. GPUCanvasContextCocoa::configure() then calls PresentationContext::configure(). This is where all the
details of the texture objects get passed in. At this point, PresentationContext creates internal
textures wrapping each of the IOSurfaces, and retains them.

Because we eagerly create the textures, most of the rest of PresentationContext is very simple; all it
has to do is track which texture is the current one.

Similarly, because the web process receives all of the backing IOSurfaces ahead of time when the canvas
got reconfigured, GPUCanvasContext doesn't have to do much beyond tracking which IOSurface is the
current one.

GPUCanvasContextCocoa::prepareForDisplay() calls CompositorIntegration::prepareForDisplay() which just
calls PresentationContext::present(). This is not quite enough, because prepareForDisplay() needs to
synchronously wait until all work has been scheduled, so we don't insert an incomplete frame into the
CoreAnimation layer tree. This improvement is being tracked at
https://bugs.webkit.org/show_bug.cgi?id=250993.

This patch deletes PresentationConfiguration, because it now is exactly identical to CanvasConfiguration,
and there's no need to have 2 configuration structs with the same contents and same purpose.

There's only really one more piece to this WebGPU compositing work, and that is to use
PresentationContext::getCurrentTexture() instead of Device::createSurfaceTexture(). I'll do that in an
upcoming patch.

* Source/WebCore/Modules/WebGPU/GPUCompositorIntegration.cpp:
(WebCore::GPUCompositorIntegration::recreateRenderBuffers const):
(WebCore::GPUCompositorIntegration::prepareForDisplay):
(WebCore::GPUCompositorIntegration::getRenderBuffers const): Deleted.
* Source/WebCore/Modules/WebGPU/GPUCompositorIntegration.h:
* Source/WebCore/Modules/WebGPU/GPUPresentationConfiguration.h: Removed.
* Source/WebCore/Modules/WebGPU/GPUPresentationContext.cpp:
(WebCore::GPUPresentationContext::configure):
* Source/WebCore/Modules/WebGPU/GPUPresentationContext.h:
* Source/WebCore/PAL/PAL.xcodeproj/project.pbxproj:
* Source/WebCore/PAL/pal/CMakeLists.txt:
* Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUCompositorIntegrationImpl.cpp:
(PAL::WebGPU::CompositorIntegrationImpl::prepareForDisplay):
(PAL::WebGPU::CompositorIntegrationImpl::recreateRenderBuffers):
(PAL::WebGPU::CompositorIntegrationImpl::getRenderBuffers): Deleted.
(PAL::WebGPU::CompositorIntegrationImpl::recreateIOSurfaces): Deleted.
* Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUCompositorIntegrationImpl.h:
* Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUConvertToBackingContext.h:
* Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUImpl.cpp:
(PAL::WebGPU::GPUImpl::createPresentationContext):
* Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUPresentationContextImpl.cpp:
(PAL::WebGPU::PresentationContextImpl::configure):
(PAL::WebGPU::PresentationContextImpl::unconfigure):
* Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUPresentationContextImpl.h:
* Source/WebCore/PAL/pal/graphics/WebGPU/WebGPUCompositorIntegration.h:
* Source/WebCore/PAL/pal/graphics/WebGPU/WebGPUPresentationConfiguration.h: Removed.
* Source/WebCore/PAL/pal/graphics/WebGPU/WebGPUPresentationContext.h:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/html/canvas/GPUCanvasContextCocoa.cpp:
(WebCore::getCanvasWidth):
(WebCore::getCanvasHeight):
(WebCore::GPUCanvasContextCocoa::GPUCanvasContextCocoa):
(WebCore::GPUCanvasContextCocoa::reshape):
(WebCore::GPUCanvasContextCocoa::configure):
(WebCore::GPUCanvasContextCocoa::unconfigure):
(WebCore::GPUCanvasContextCocoa::getCurrentTexture):
(WebCore::GPUCanvasContextCocoa::prepareForDisplay):
(WebCore::getCanvasSizeAsIntSize): Deleted.
(WebCore::GPUCanvasContextCocoa::unconfigurePresentationContextIfNeeded): Deleted.
(WebCore::GPUCanvasContextCocoa::configurePresentationContextIfNeeded): Deleted.
* Source/WebCore/html/canvas/GPUCanvasContextCocoa.h:
* Source/WebGPU/WebGPU/PresentationContext.h:
* Source/WebGPU/WebGPU/PresentationContext.mm:
(WebGPU::PresentationContext::unconfigure):
(wgpuSwapChainRelease):
* Source/WebGPU/WebGPU/PresentationContextCoreAnimation.h:
* Source/WebGPU/WebGPU/PresentationContextCoreAnimation.mm:
(WebGPU::PresentationContextCoreAnimation::unconfigure):
* Source/WebGPU/WebGPU/PresentationContextIOSurface.h:
(WebGPU::PresentationContextIOSurface::create): Deleted.
(WebGPU::PresentationContextIOSurface::displayBuffer const): Deleted.
(WebGPU::PresentationContextIOSurface::drawingBuffer const): Deleted.
* Source/WebGPU/WebGPU/PresentationContextIOSurface.mm:
(WebGPU::PresentationContextIOSurface::create):
(WebGPU::PresentationContextIOSurface::PresentationContextIOSurface):
(WebGPU::PresentationContextIOSurface::displayBuffer const):
(WebGPU::PresentationContextIOSurface::drawingBuffer const):
(WebGPU::PresentationContextIOSurface::renderBuffersWereRecreated):
(WebGPU::PresentationContextIOSurface::configure):
(WebGPU::PresentationContextIOSurface::unconfigure):
(WebGPU::PresentationContextIOSurface::present):
(WebGPU::PresentationContextIOSurface::getCurrentTexture):
(WebGPU::PresentationContextIOSurface::getCurrentTextureView):
(wgpuSurfaceCocoaCustomSurfaceGetDisplayBuffer):
(wgpuSurfaceCocoaCustomSurfaceGetDrawingBuffer):
(WebGPU::optionsFor32BitSurface): Deleted.
(WebGPU::createIOSurface): Deleted.
(WebGPU::createSurfaceFromDescriptor): Deleted.
(WebGPU::PresentationContextIOSurface::nextDrawable): Deleted.
* Source/WebGPU/WebGPU/Texture.h:
* Source/WebGPU/WebGPU/Texture.mm:
(WebGPU::Texture::usage):
(WebGPU::Device::createTexture):
(WebGPU::usage): Deleted.
* Source/WebGPU/WebGPU/WebGPUExt.h:
* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteCompositorIntegration.cpp:
(WebKit::RemoteCompositorIntegration::recreateRenderBuffers):
(WebKit::RemoteCompositorIntegration::prepareForDisplay):
(WebKit::RemoteCompositorIntegration::getRenderBuffers): Deleted.
* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteCompositorIntegration.h:
* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteCompositorIntegration.messages.in:
* Source/WebKit/GPUProcess/graphics/WebGPU/RemotePresentationContext.cpp:
(WebKit::RemotePresentationContext::configure):
* Source/WebKit/GPUProcess/graphics/WebGPU/RemotePresentationContext.h:
* Source/WebKit/GPUProcess/graphics/WebGPU/RemotePresentationContext.messages.in:
* Source/WebKit/Scripts/webkit/messages.py:
(headers_for_type):
* Source/WebKit/Shared/WebGPU/WebGPUConvertFromBackingContext.h:
* Source/WebKit/Shared/WebGPU/WebGPUConvertToBackingContext.h:
* Source/WebKit/Shared/WebGPU/WebGPUPresentationConfiguration.cpp: Removed.
* Source/WebKit/Shared/WebGPU/WebGPUPresentationConfiguration.h: Removed.
* Source/WebKit/Shared/WebGPU/WebGPUPresentationConfiguration.serialization.in: Removed.
* Source/WebKit/Sources.txt:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteCompositorIntegrationProxy.cpp:
(WebKit::WebGPU::RemoteCompositorIntegrationProxy::recreateRenderBuffers):
(WebKit::WebGPU::RemoteCompositorIntegrationProxy::prepareForDisplay):
(WebKit::WebGPU::RemoteCompositorIntegrationProxy::getRenderBuffers): Deleted.
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteCompositorIntegrationProxy.h:
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemotePresentationContextProxy.cpp:
(WebKit::WebGPU::RemotePresentationContextProxy::configure):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemotePresentationContextProxy.h:

Canonical link: https://commits.webkit.org/259778@main
